src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -1018,14 +1110,29 @@ public synchronized void renderGraph(Graphics g) {
                 g.setColor(Colors.LIVE_CSET);
                 g.drawRect(x, (int) Math.round(startRaw - s.collectionSet() * stepY), 1, 1);
 
-                // Draw this in the lower band.
-                final int smooth = Math.min(10, i + 1);
-                final int mult = 50;
-
-                SnapshotView ls = lastSnapshots.get(i - smooth + 1);
-
-                g.setColor(Colors.USED);
-                g.drawRect(x, (int) Math.round(startDiff - (s.used() - ls.used()) * stepY * mult / smooth), 1, 1);
+                g.setColor(Color.GRAY);
+                g.drawString("OM", bandWidth + 10, bandHeight + pad + 20);
+                g.drawString("M", bandWidth + 10, bandHeight + phaseHeight + pad + 20);
+                g.drawString("E", bandWidth + 10, bandHeight + 2*phaseHeight + pad + 20);
+                g.drawString("UR", bandWidth + 10, bandHeight + 3*phaseHeight + pad + 20);
+
+
+                Graphics2D g2 = (Graphics2D) g;
+                g2.setStroke(new BasicStroke(2));
+                g2.drawLine(0, bandHeight + 5, 0, bandHeight + pad - 5);
+                g2.drawLine(bandWidth / 4, bandHeight + 5, bandWidth / 4, bandHeight + pad - 5);
+                g2.drawLine(bandWidth / 2, bandHeight + 5, bandWidth / 2, bandHeight + pad - 5);
+                g2.drawLine(bandWidth * 3 / 4, bandHeight + 5, bandWidth * 3 / 4, bandHeight + pad - 5);
+                g.drawString("-" + Long.toString(popupSnapshots.get(popupSnapshots.size() - 1).time() - popupSnapshots.get(frontSnapshotIndex).time()) + " ms", 3, bandHeight + 20);
```

Comment:
This looks a lot like the code back on line 810 (ish). Can we deduplicate this?

Reply:
For the 810(ish) code is the part for the Live mode and here is for the Replay mode. Because two modes have different snapshots data so they look really similar but they are handling the data differently.